app/models/dialog_field_text_box.rb

```diff
@@ -1,9 +1,10 @@
 class DialogFieldTextBox < DialogField
-  AUTOMATE_VALUE_FIELDS = %w(protected required validator_rule validator_type read_only)
+  AUTOMATE_VALUE_FIELDS = %w(data_type protected required validator_rule validator_type read_only).freeze
 
   def value
     @value = values_from_automate if dynamic && @value.blank?
-    @value
+    return @value if @value.nil?
+    data_type == "integer" ? @value.to_i : @value
```

Comment:
you can combine two lines into just one
`data_type == 'integer' ? @value.to_i : @value if @value`